content/post/event-triggered-customer-segmentation.md

```diff
@@ -0,0 +1,247 @@
++++
+title = "Event Triggered Customer Segmentation"
+date = 2018-07-17T10:40:34+02:00
+draft = false
+author = "rblasucci"
+description = "In which we quickly and easily create a full application to use Wallaroo to manage an ad campaign for a Marsupial Fan Club."
+tags = [
+    "Python",
+    "State",
+    "Wallaroo",
+]
+categories = [
+    "Python-Api",
+    "State",
+    "Stream-Processing",
+]
++++
+
+Today I'm going to show you how fast and easy it can be to set up a simple application with Wallaroo to manage an ad campaign.
+
+## Backstory
+You’re a data engineer getting hounded by the marketing team at the fictional online Marsupial Fan Club to support an online ad campaign they have conjured up with a goal of converting an ad for a hoodie with an adorable marsupial on the front to purchases..  The Marsupial-Fan-Club has a single mission - market marsupial chatzky to the adoring fans.  The marketing channels are Instagram and Facebook -  dedicated to adorable pictures of marsupials of all flavors and kinds. You support the business through  running a modest storefront with assorted items for sale -- coffee cups, hoodies, and the like. For the Marsupial aficionados, you also have a special loyalty-members-only list where you send out additional coupons and special content. Today, you'd like to run a promotion for 10% off your newest hoodie. But there's a twist! If one of your known loyalty members clicks on your ad, you want to send them a coupon for an additional 10% off. With Wallaroo, this is super easy! Let's take an in-depth look at how to get there.
+
+## Data structures
+First, let's consider what our data looks like, so you'll know what pieces of information you'll need to consider.
+
+### Loyalty list
+Most importantly, we have our loyalty list. Today, we're sending in a text file populated with non-existing email accounts (to ensure we don't use someone's real address), with each line simulating a new entry. This could be easily be data that's read from another source, instead. We'll be tracking emails, Facebook username, Instagram username, and their gender. You might also track things like date of last fanclub purchase, geography, age demographics, # of insta followers (to understand their social influence/reach), etc.
+```
+kangak@fakeemail.com, kangakerfluffle, kangakerfluffle, F
+wackywallaroo@fakeemail.com, , wackywallaroo, M
+...
+```
+This will easily convert into a python object, like so: 
+```
```

Comment:
Put `python` after the three backticks so that Python syntax highlighting is used.